File: ClientApp/app/auth/privilege.ts
import { Injectable } from "@angular/core";

@Injectable()
export class Privilege {

    static General = class {
        static Unknown: string = "unknown";
        static UpdateOnderneming: string = "updateonderneming";
        static Nachtverwerking: string = "nachtverwerking";
    }

    static AbonnementDieptekavelWijziging = class {
        static GetByAbonnementId: string = "getdieptekavelwijzigingenbyabonnementid";
    }

    static JuridischeEntiteit = class {
        static Get: string = "getjuridischeentiteit";
    }

    static Beheer = class {
        static BeheerMenu: string = "beheer";
    }

    static Rapportage = class {
        static Get: string = "getrapportage";
    }

    static CodeTable = class {
        static Get: string = "getcodetable";
    }

    static AbonnementStatus = class {
        static ChangeStatus: string = "changeabonnementstatus";
        static AddWijziging: string = "addabonnementstatuswijziging";
    }

    static AbonnementElektriciteitwijziging = class {
        static Add: string = "addabonnementelektriciteitwijziging";
        static Update: string = "updateabonnementelektriciteitwijziging";
        static Delete: string = "deleteabonnementelektriciteitwijziging";
        static GetByAbonnementId: string = "geteelektriciteitwijzigingenbyabonnementid";
    }

    static Klant = class {
        static GetCommerciele: string = "getcommercieleklanten";
        static GetNietCommerciele: string = "getnietcommercieleklanten";
        static GetByOndernemingsnr: string = "getklantbyondernemingsnr";
        static GetAll: string = "getallklanten";
        static GetAllFilteredAndPaged: string = "getallklantenfilteredandpaged";
        static Get: string = "getklant";
        static Add: string = "addklant";
        static Update: string = "updateklant";
        static Delete: string = "deleteklant";
        static Select: string = "selectklant";
    }

    static Markt = class {
        static Get: string = "getmarkt";
        static GetMarktBoekjes: string = "getmarktboekjes";
        static GetAll: string = "getallmarkten";
        static GetAllFilteredAndPaged: string = "getallmarktenfilteredandpaged";
        static GetByDistrictid: string = "getmarktenbydistrictid";
        static Add: string = "addmarkt";
        static Update: string = "updatemarkt";
        static Delete: string = "deletemarkt";
    }

    static AanvraagMarkt = class {
        static Get: string = "getaanvraagmarkt";
        static Add: string = "addaanvraagmarkt";
        static Update: string = "updateaanvraagmarkt";
        static Delete: string = "deleteaanvraagmarkt";
        static GetByKlantId: string = "getmarktaanvragenbyklantid";
    }

    static Aanvraag = class {
        static Get: string = "getaanvraag";
        static GetAll: string = "getallaanvragen";
        static GetAllFilteredAndPaged: string = "getallaanvragenfilteredandpaged";
        static GetAllByStatusFilteredAndPaged: string = "getallaanvragenbystatusfilteredandpaged";
        static Add: string = "addaanvraag";
        static Update: string = "updateaanvraag";
        static Delete: string = "deleteaanvraag";
        static Detail: string = "intakedetail";
        static CreateForKlant: string = "createintakeforklant";
        static VeranderNaarGepreregistreerd: string = "veranderintakenaargepreregistreerd";
        static ReservatieRaadplegen: string = "reservatieraadplegen";
        static RegistratieBewerken: string = "registratiebewerken";
        static RegistratieVerwijderen: string = "registratieverwijderen";
        static ToewijzenAanvragenNietInAanmerking: string = "toewijzenaanvragennietinaanmerking";
        static PreregistratieBewerken: string = "preregistratiebewerken";
        static DirectRegistreren: string = "directregistreren";
    }

    static Abonnement = class {
        static Get: string = "getabonnement";
        static GetAll: string = "getallabonnementen";
        static GetAllFilteredAndPaged: string = "getallabonnementenfilteredandpaged";
        static GetAllByStatusFilteredAndPaged: string = "getallabonnementenbystatusfilteredandpaged";
        static Add: string = "addabonnement";
        static Update: string = "updateabonnement";
        static Delete: string = "deleteabonnement";
        static Detail: string = "abonnementdetail";
        static GetByKlantId: string = "getabonnementenbyklantid";
    }

    static Kavel = class {
        static Get: string = "getkavel";
        static Update: string = "updatekavel";
        static Delete: string = "deletekavel";
        static Add: string = "addkavel";
    }

    static KavelOpmerking = class {
        static Get: string = "getkavelopmerking";
        static Add: string = "addkavelopmerking";
        static Update: string = "updatekavelopmerking";
        static Delete: string = "deletekavelopmerking";
        static Toevoegen: string = "kavelopmerkingtoevoegen";
    }

    static Aanvraagweigering = class {
        static GetByAanvraagId: string = "getaanvraagweigeringbyaanvraagid";
        static Add: string = "addaanvraagweigering";
        static Get: string = "getaanvraagweigering";
        static Update: string = "updateaanvraagweigering";
        static Delete: string = "deleteaanvraagweigering";
    }

    static Factuur = class {
        static Get: string = "getfactuur";
        static GetAll: string = "getallfacturen";
        static GetFilteredAndPaged: string = "getfacturenfilteredandpaged";
        static Add: string = "addfactuur";
        static Update: string = "updatefactuur";
        static Delete: string = "deletefactuur";
        static Genereer: string = "genereerfacturen";
        static StuurNaarSap: string = "stuurfacturennaarsap";
        static FacturatieMenu: string = "facturatiemenu";
    }

    static Document = class {
        static Get: string = "getdocument";
        static Add: string = "adddocument";
        static Update: string = "updatedocument";
        static Delete: string = "deletedocument";
    }

    static Locatie = class {
        static Get: string = "getlocatie";
        static Add: string = "addlocatie";
        static Update: string = "updatelocatie";
        static Delete: string = "deletelocatie";
    }

    static Product = class {
        static Get: string = "getproduct";
        static Add: string = "addproduct";
        static Update: string = "updateproduct";
        static Delete: string = "deleteproduct";
    }

    static AbonnementKavel = class {
        static Add: string = "addabonnementkavel";
        static Update: string = "updateabonnementkavel";
        static Delete: string = "deleteabonnementkavel";
        static Get: string = "getabonnementkavelabonnementkavel";
    }

    static District = class {
        static Get: string = "getdistrict";
        static Add: string = "adddistrict";
        static Update: string = "updatedistrict";
        static Delete: string = "deletedistrict";
    }
}

File: Shared/Privilege.cs
namespace MarktenEnForenApi.Shared.Authentication
{
    public class Privilege
    {
        public class General
        {
            public const string Unknown = "unknown";
            public const string UpdateOnderneming = "updateonderneming";
            public const string Nachtverwerking = "nachtverwerking";
        }

        public class AbonnementDieptekavelWijziging
        {
            public const string GetByAbonnementId = "getdieptekavelwijzigingenbyabonnementid";
        }

        public class JuridischeEntiteit
        {
            public const string Get = "getjuridischeentiteit";
        }

        public class Rapportage
        {
            public const string Get = "getrapportage";
        }

        public class CodeTable
        {
            public const string Get = "getcodetable";
        }

        public class AbonnementStatus
        {
            public const string ChangeStatus = "changeabonnementstatus";
            public const string AddWijziging = "addabonnementstatuswijziging";
        }

        public class AbonnementElektriciteitwijziging
        {
            public const string Add = "addabonnementelektriciteitwijziging";
            public const string Update = "updateabonnementelektriciteitwijziging";
            public const string Delete = "deleteabonnementelektriciteitwijziging";
            public const string GetByAbonnementId = "geteelektriciteitwijzigingenbyabonnementid";
        }

        public class Klant
        {
            public const string GetCommerciele = "getcommercieleklanten";
            public const string GetNietCommerciele = "getnietcommercieleklanten";
            public const string GetByOndernemingsnr = "getklantbyondernemingsnr";
            public const string GetAll = "getallklanten";
            public const string GetAllFilteredAndPaged = "getallklantenfilteredandpaged";
            public const string Get = "getklant";
            public const string Add = "addklant";
            public const string Update = "updateklant";
            public const string Delete = "deleteklant";
            public const string Select = "selectklant";
        }

        public class Markt
        {
            public const string Get = "getmarkt";
            public const string GetMarktBoekjes = "getmarktboekjes";
            public const string GetAll = "getallmarkten";
            public const string GetAllFilteredAndPaged = "getallmarktenfilteredandpaged";
            public const string GetByDistrictid = "getmarktenbydistrictid";
            public const string Add = "addmarkt";
            public const string Update = "updatemarkt";
            public const string Delete = "deletemarkt";
        }

        public class AanvraagMarkt
        {
            public const string Get = "getaanvraagmarkt";
            public const string Add = "addaanvraagmarkt";
            public const string Update = "updateaanvraagmarkt";
            public const string Delete = "deleteaanvraagmarkt";
            public const string GetByKlantId = "getmarktaanvragenbyklantid";
        }

        public class Aanvraag
        {
            public const string Get = "getaanvraag";
            public const string GetAll = "getallaanvragen";
            public const string GetAllFilteredAndPaged = "getallaanvragenfilteredandpaged";
            public const string GetAllByStatusFilteredAndPaged = "getallaanvragenbystatusfilteredandpaged";
            public const string Add = "addaanvraag";
            public const string Update = "updateaanvraag";
            public const string Delete = "deleteaanvraag";
            public const string Detail = "intakedetail";
            public const string CreateForKlant = "createintakeforklant";
            public const string VeranderNaarGepreregistreerd = "veranderintakenaargepreregistreerd";
            public const string ReservatieRaadplegen = "reservatieraadplegen";
            public const string RegistratieBewerken = "registratiebewerken";
            public const string RegistratieVerwijderen = "registratieverwijderen";
            public const string ToewijzenAanvragenNietInAanmerking = "toewijzenaanvragennietinaanmerking";
            public const string PreregistratieBewerken = "preregistratiebewerken";
            public const string DirectRegistreren = "directregistreren";
        }

        public class Abonnment
        {
            public const string Get = "getabonnement";
            public const string GetAll = "getallabonnementen";
            public const string GetAllFilteredAndPaged = "getallabonnementenfilteredandpaged";
            public const string GetAllByStatusFilteredAndPaged = "getallabonnementenbystatusfilteredandpaged";
            public const string Add = "addabonnement";
            public const string Update = "updateabonnement";
            public const string Delete = "deleteabonnement";
            public const string Detail = "abonnementdetail";
            public const string GetByKlantId = "getabonnementenbyklantid";
        }

        public class Kavel
        {
            public const string Get = "getkavel";
            public const string Update = "updatekavel";
            public const string Delete = "deletekavel";
            public const string Add = "addkavel";
        }

        public class KavelOpmerking
        {
            public const string Get = "getkavelopmerking";
            public const string Add = "addkavelopmerking";
            public const string Update = "updatekavelopmerking";
            public const string Delete = "deletekavelopmerking";
            public const string Toevoegen = "kavelopmerkingtoevoegen";
        }

        public class Aanvraagweigering
        {
            public const string GetByAanvraagId = "getaanvraagweigeringbyaanvraagid";
            public const string Add = "addaanvraagweigering";
            public const string Get = "getaanvraagweigering";
            public const string Update = "updateaanvraagweigering";
            public const string Delete = "deleteaanvraagweigering";
        }

        public class Factuur
        {
            public const string Get = "getfactuur";
            public const string GetAll = "getallfacturen";
            public const string GetFilteredAndPaged = "getfacturenfilteredandpaged";
            public const string Add = "addfactuur";
            public const string Update = "updatefactuur";
            public const string Delete = "deletefactuur";
            public const string Genereer = "genereerfacturen";
            public const string StuurNaarSap = "stuurfacturennaarsap";
            public const string FacturatieMenu = "facturatiemenu";
        }

        public class Document
        {
            public const string Get = "getdocument";
            public const string Add = "adddocument";
            public const string Update = "updatedocument";
            public const string Delete = "deletedocument";
        }

        public class Locatie
        {
            public const string Get = "getlocatie";
            public const string Add = "addlocatie";
            public const string Update = "updatelocatie";
            public const string Delete = "deletelocatie";
        }

        public class Product
        {
            public const string Get = "getproduct";
            public const string Add = "addproduct";
            public const string Update = "updateproduct";
            public const string Delete = "deleteproduct";
        }

        public class AbonnementKavel
        {
            public const string Add = "addabonnementkavel";
            public const string Update = "updateabonnementkavel";
            public const string Delete = "deleteabonnementkavel";
            public const string Get = "getabonnementkavel";
        }

        public class District
        {
            public const string Get = "getdistrict";
            public const string Add = "adddistrict";
            public const string Update = "updatedistrict";
            public const string Delete = "deletedistrict";
        }
    }
}
